packages/config/src/load.ts

```diff
@@ -0,0 +1,53 @@
+import type { ConfigResult, FlatbreadConfig } from '@flatbread/core';
+import {
+  validateConfigHasDefaultExport,
+  validateConfigStructure,
+} from './validate';
+import { cosmiconfig } from 'cosmiconfig';
+
+/**
+ * Loads an ESModule-style config file.
+ */
+const esmLoader = async (filePath: string) => {
+  const configModule = await import(filePath);
+
+  validateConfigHasDefaultExport(configModule);
+  validateConfigStructure(configModule.default);
+
+  return configModule.default;
+};
+
+/**
+ * Pulls the user config from an optionally specified filepath.
+ *
+ * By default, this will search the current working directory.
+ *
+ * @param options options for loading the config file, defaults to `{}`. Can pass in `cwd` as a path `string` to override the current working directory.
+ * @returns Promise that resolves to the user config object.
+ */
+export async function loadConfig({ cwd = process.cwd() } = {}): Promise<
+  ConfigResult<FlatbreadConfig>
+> {
+  const moduleName = 'flatbread';
+
+  const validExtensions = ['.js', '.mjs'];
+  const configFileNamingScheme = [`.${moduleName}rc`, `${moduleName}.config`];
+  const configFileNamePermutations = configFileNamingScheme.flatMap((scheme) =>
+    validExtensions.map((ext) => `${scheme}${ext}`)
+  );
+
+  const explorer = cosmiconfig('flatbread', {
+    searchPlaces: configFileNamePermutations,
+    loaders: {
+      '.js': esmLoader,
+      '.mjs': esmLoader,
```

Comment:
for example, we could add `'.ts': tsxLoader`, using https://github.com/esbuild-kit/tsx and probably get around the issue I was having with using `ts-node`

Reply:
When I tried the `.ts` config file it exposed a bunch of issues in Flatbread's config types we should correct - so I think it's important to have as an option even from the standpoint of integration testing

Reply:
I agree, I think I screwed up some of the types with the overrides merge